samples/snippets/src/main/java/com/example/bigquery/QueryExternalBigtableTemp.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.bigquery;
+
+// [START bigquery_query_external_bigtable_temp]
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryException;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.BigtableColumn;
+import com.google.cloud.bigquery.BigtableColumnFamily;
+import com.google.cloud.bigquery.BigtableOptions;
+import com.google.cloud.bigquery.ExternalTableDefinition;
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.TableResult;
+import com.google.common.collect.ImmutableList;
+import org.apache.commons.codec.binary.Base64;
+
+// Sample to queries an external bigtable data source using a temporary table
+public class QueryExternalBigtableTemp {
+
+  public static void main(String[] args) {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "MY_PROJECT_ID";
+    String instanceId = "MY_INSTANCE_ID";
+    String bigtableName = "MY_BIGTABLE_NAME";
+    String tableName = "MY_TABLE_NAME";
+    String sourceUri =
+        "https://googleapis.com/bigtable/projects/"
+            + projectId
+            + "/instances/"
+            + instanceId
+            + "/tables/"
+            + bigtableName;
+    String query = String.format("SELECT * FROM %s ", tableName);
+    queryExternalBigtableTemp(tableName, sourceUri, query);
+  }
+
+  public static void queryExternalBigtableTemp(String tableName, String sourceUri, String query) {
+    try {
+      // Initialize client that will be used to send requests. This client only needs to be created
+      // once, and can be reused for multiple requests.
+      BigQuery bigquery = BigQueryOptions.getDefaultInstance().getService();
+
+      BigtableColumn name =
+          BigtableColumn.newBuilder()
+              .setQualifierEncoded(Base64.encodeBase64String("name".getBytes()))
+              .setFieldName("name")
+              .setType("STRING")
+              .setEncoding("TEXT")
+              .build();
+      BigtableColumn postAbbr =
+          BigtableColumn.newBuilder()
+              .setQualifierEncoded(Base64.encodeBase64String("post_abbr".getBytes()))
+              .setFieldName("post_abbr")
+              .setType("STRING")
+              .setEncoding("TEXT")
+              .build();
+      BigtableColumnFamily usStates =
```

Comment:
Could you define the column family first and then add the columns and build after? It will make more sense for someone from Bigtable following since column families are the higher level grouping

Reply:
@billyjacobson It's not possible to main order, because first we need to create an object of `BigtableColumn` then we can set those columns in `BigtableColumnFamily`.

Reply:
we could first instantiate an empty `BigtableColumnFamily` object and then add columns to it.